src/main/java/uk/gov/companieshouse/company/metrics/service/NullMetricsClient.java

```diff
@@ -8,7 +8,7 @@
 import uk.gov.companieshouse.logging.Logger;
 import uk.gov.companieshouse.logging.LoggerFactory;
 
-@Component
+@Component("nullMetricsClient")
```

Comment:
Why did we add names for these components? I think we normally do this to avoid bean definition conflicts but I cant see anywhere where we have specifically targeted these beans. Do we defo need these? :)

Reply:
Needed this because the service passes around and defines those clients polymorphically via the parent class so when spring was attempting to autowire those clients in it would throw the "expected 1 bean but found x"